### PR TITLE
Main for requests

### PR DESCRIPTION
**Pull request by dliaudan**
Hi, here is created all task that you asked me to do, including:
1. Created modules for EC2 and SG (I make them flexible as much as it needed for current purposes)
2. Created resource for ELB
3. Created iam policy for SSM, iam role attach it together and attached it to instance profile
4. Create bash script that installs on Ubuntu aws-cli and put from SSM secret with username:password credentials and save it into /tmp/credentials.env file

Please check my project. I'll be appreciate to make improvments to my code, making it more flexible and pure :)

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -69,60 +69,126 @@ resource "aws_route_table_association" "this" {
...
-  vpc_id      = aws_vpc.this.id
-#each ingress means one ingress rule
-  ingress {
-    description      = "HTTPS from EC2 instance"
-    from_port        = 443
-    to_port          = 443
-    protocol         = "tcp"
-    cidr_blocks      = ["0.0.0.0/0"]
-  }
+module "Security_group_EC2" {
```
Comment (koshmanov) [inline main.tf:72]: by best practices you have to use only lower registry in naming

File: .gitignore [hunk 2]
```diff
@@ -35,5 +35,4 @@ override.tf.json
 terraform.rc
 
 
-variables.tf
 *.tfvars
```
Comment (koshmanov) [inline .gitignore:38]: this part should not be in gitignore section

File: main.tf [hunk 3]
```diff
@@ -69,60 +69,126 @@ resource "aws_route_table_association" "this" {
...
+  source          = ".//modules/Security_group"
+  vpc_id          = "${aws_vpc.this.id}"
```
Comment (koshmanov) [inline main.tf:74]: Try to validate your code via terraform command. I guess it should throw some warning here :)

File: main.tf [hunk 4]
```diff
@@ -69,60 +69,126 @@ resource "aws_route_table_association" "this" {
...
+  name               = "EC2_role_for_SM"
+  path               = "/"
+  description        = "Policy for read/write data on secret manager (SM)" 
+  policy             = jsonencode({
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": [
+                "secretsmanager:*",
+                "cloudformation:CreateChangeSet",
```
Comment (koshmanov) [inline main.tf:91]: Why do you need access to `cloudformation` ?

File: main.tf [hunk 5]
```diff
@@ -69,60 +69,126 @@ resource "aws_route_table_association" "this" {
...
+            "Effect": "Allow",
+            "Resource": "arn:aws:serverlessrepo:*:*:applications/SecretsManager*"
+        },
+        {
+            "Action": [
+                "s3:GetObject"
+            ],
+            "Effect": "Allow",
+            "Resource": [
+                "arn:aws:s3:::awsserverlessrepo-changesets*",
```
Comment (koshmanov) [inline main.tf:136]: :)

File: main.tf [hunk 6]
```diff
@@ -69,60 +69,126 @@ resource "aws_route_table_association" "this" {
...
 
-  egress {
-    from_port        = 0
-    to_port          = 0
-    protocol         = "-1"
-    cidr_blocks      = ["0.0.0.0/0"]
-  }
+  # Terraform's "jsonencode" function converts a
+  # Terraform expression result to valid JSON syntax.
+  assume_role_policy = jsonencode({
```
Comment (koshmanov) [inline main.tf:151]: What does `assume_role_policy` field means for role?

File: main.tf [hunk 7]
```diff
@@ -69,60 +69,126 @@ resource "aws_route_table_association" "this" {
...
+resource "aws_iam_instance_profile" "this" {
+  name = "ec2_sm_profile"
+  role = aws_iam_role.this.name
+}
 
-  tags = {
-    Name = "Terraform server"
-  }
+#creating EC2 instance 
+module "module_server_test" {
```
Comment (koshmanov) [inline main.tf:182]: SG groups works tightly with EC2 server. You can put these resources into one module. 
The idea of modules is to simplify the code.

File: main.tf [hunk 8]
```diff
@@ -136,7 +202,7 @@ resource "aws_ebs_volume" "this" {
 resource "aws_volume_attachment" "this" {
```
Comment (koshmanov) [inline main.tf:202]: Why didn't you move this part into module?

File: modules/Security_group/outputs.tf [hunk 9]
```diff
@@ -0,0 +1,4 @@
+output "Security_group_id" {
```
Comment (koshmanov) [inline modules/Security_group/outputs.tf:1]: naming convention.

**Review by koshmanov — CHANGES_REQUESTED** [review]
Please update the resources names and fix the instance profile policy.

File: main.tf (hunk 3 above)
Comment (dliaudan) [reply main.tf:74]: image
Tried, there are no warnings, what here should be?

File: main.tf (hunk 7 above)
Comment (dliaudan) [reply main.tf:182]: Yes, but here is an another task for creating SG on LB. In this case - I should create separate SG resource for LB?

File: main.tf (hunk 3 above)
Comment (koshmanov) [reply main.tf:74]: image
You can pass value without using quotes ("")

File: main.tf [hunk 10]
```diff
@@ -69,74 +69,81 @@ resource "aws_route_table_association" "this" {
...
+                "sts:AssumeRole" #action is assuming role
+            ],
+            "Principal": { #fspecify for whom permission will be granted
+                "Service": [ #define which services will be impacted for assuming (also here can be accounts)
+                    "ec2.amazonaws.com" #especially assuming this role is allowed for ec2 instances
+                ]
+            }
+        }
+    ]
+}) #assume role defines for which subjects (accounts or instances) this role can be applied
```
Comment (koshmanov) [inline main.tf:119]: 👍

**Review by koshmanov — APPROVED** [review]
Good job 👍

**Pull request merged**